sandbox/apparmor/notify/version.go

```diff
@@ -0,0 +1,82 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2025 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package notify
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/testutil"
+)
+
+// ProtocolVersion denotes a notification protocol version.
+type ProtocolVersion uint16
+
+// versions holds the notification protocols snapd supports, in order of
+// preference. If the first version is supported, try to use it, else try the
+// next version, etc.
+var (
+	versions = []ProtocolVersion{3}
+
+	versionSupportedCallbacks = map[ProtocolVersion]func() bool{
```

Comment:
shouldn't this be called more something like versionSupportedByKernelHintCheck ?